### PR TITLE
enh(auth): refactor authentication endpoints

### PR DESCRIPTION
**Pull request by wtermellil**
## Description

**PLEASE MAKE SURE THAT THE BRANCH PR INCLUDES JIRA TICKET ID** (_for centreon-internal_)

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.04.x
- [ ] 21.10.x
- [ ] 22.04.x
- [x] 22.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by jeremyjaouen — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php [hunk 1]
```diff
@@ -467,31 +489,35 @@ private function createAuthenticationTokens(): void
...
-                    'verify_peer' => $this->configuration->verifyPeer()
+                    'verify_peer' => $customConfiguration->verifyPeer()
                 ]
             );
-        } catch (\Exception $ex) {
-            $this->logExceptionInLoginLogFile("Unable to get Introspection Information: %s, message: %s", $ex);
+        } catch (Exception $e) {
+            $this->logExceptionInLoginLogFile("Unable to get Introspection Information: %s, message: %s", $e);
             $this->error(sprintf(
                 "[Error] Unable to get Token Introspection Information:, message: %s",
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Domain/Provider/OpenIdProvider.php]:
```suggestion
                "[Error] Unable to get Introspection Token Information:, message: %s",
```

File: src/Core/Security/Authentication/Infrastructure/Provider/WebSSO.php [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+    /**
+     * Validate that login attribute is defined in server environment variables
+     */
+    public function validateLoginAttributeOrFail(): void
+    {
+        /** @var CustomConfiguration $customConfiguration */
+        $customConfiguration = $this->getConfiguration()->getCustomConfiguration();
+        $this->info('Validating login header attribute');
+        if (!array_key_exists($customConfiguration->getLoginHeaderAttribute(), $_SERVER)) {
+            $this->error('login header attribute not found in server environment server', [
```
Comment (cg-tw) [inline src/Core/Security/Authentication/Infrastructure/Provider/WebSSO.php]: server environment server?

File: src/EventSubscriber/WebSSOEventSubscriber.php [hunk 3]
```diff
@@ -335,7 +227,7 @@ private function createAuthenticationTokens(
             if (!$isAlreadyInTransaction) {
                 $this->dataStorageEngine->commitTransaction();
             }
-        } catch (\Exception $ex) {
+        } catch (Exception $ex) {
             $this->error('Unable to create authentication tokens', [
```
Comment (cg-tw) [inline src/EventSubscriber/WebSSOEventSubscriber.php:231]: token or tokens?

File: tests/php/Core/Security/AccessGroup/Domain/Model/AccessGroupTest.php [hunk 4]
```diff
@@ -21,11 +21,11 @@
...
 
-namespace Tests\Core\Security\Domain\AccessGroup\Model;
+namespace Tests\Core\Security\AccessGroup\Domain\Model;
 
 use Assert\InvalidArgumentException;
 use Centreon\Domain\Common\Assertion\AssertionException;
-use Core\Security\Domain\AccessGroup\Model\AccessGroup;
+use Core\Security\AccessGroup\Domain\Model\AccessGroup;
 
 it('should thrown an Exception when an access group name is empty', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/AccessGroup/Domain/Model/AccessGroupTest.php]:
```suggestion
it('should throw an Exception when an access group name is empty', function () {
```

File: tests/php/Core/Security/Authentication/Application/UseCase/LoginOpenIdSession/LoginOpenIdSessionTest.php [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+        'contact_group' => new ContactGroup(3, 'contact_group'),
+        'claim_name' => 'groups',
+        'authorization_rules' => [],
+
+    ]);
+    $configuration->setCustomConfiguration($customConfiguration);
+    $this->validOpenIdConfiguration = $configuration;
+});
+
+it('expects to return an error message in presenter when no provider configuration are found', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/Authentication/Application/UseCase/LoginOpenIdSession/LoginOpenIdSessionTest.php]:
```suggestion
it('expects to return an error message in presenter when no provider configuration is found', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Builder/DbConfigurationBuilderTest.php [hunk 6]
```diff
@@ -50,27 +51,40 @@
...
         'authentication_type' => 'client_secret_post',
         'verify_peer' => false,
-        'contact_group' => new ContactGroup(1, 'contact_group'),
+        'contact_group' => 1,
         'claim_name' => 'groups',
         'authorization_rules' => [],
     ];
 });
 
 it('should throw an exception when a mandatory parameters is empty and configuration is active', function () {
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Builder/DbConfigurationBuilderTest.php]:
```suggestion
it('should throw an exception when a mandatory parameter is empty and configuration is active', function () {
```

File: tests/php/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Builder/DbConfigurationBuilderTest.php [hunk 7]
```diff
@@ -50,27 +51,40 @@
...
+    $configuration = new Configuration(
+        2,
+        strtolower(Provider::OPENID),
+        Provider::OPENID,
+        json_encode($this->customConfiguration),
+        true,
+        false
+    );
+    $configuration->setCustomConfiguration(new CustomConfiguration($this->customConfiguration));
+})->throws(OpenIdConfigurationException::class, "Missing mandatory parameters: base_url");
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Builder/DbConfigurationBuilderTest.php:71]:
```suggestion
})->throws(OpenIdConfigurationException::class, "Missing mandatory parameter: base_url");
```

File: tests/php/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Builder/DbConfigurationBuilderTest.php [hunk 8]
```diff
@@ -81,12 +95,17 @@ function () {
     'should throw an exception when the configuration is active, autoimport enable but with missing parameters',
```
Comment (cg-tw) [inline tests/php/Core/Security/ProviderConfiguration/Infrastructure/OpenId/Builder/DbConfigurationBuilderTest.php]:
```suggestion
    'should throw an exception when the configuration is active, autoimport enabled but with missing parameters',
```

File: src/EventSubscriber/WebSSOEventSubscriber.php (hunk 3 above)
Comment (wtermellil) [reply src/EventSubscriber/WebSSOEventSubscriber.php:231]: tokens

File: src/Core/Security/Authentication/Infrastructure/Provider/WebSSO.php (hunk 2 above)
Comment (wtermellil) [reply src/Core/Security/Authentication/Infrastructure/Provider/WebSSO.php]:
```suggestion
            $this->error('login header attribute not found in server environment', [
```

**Review by jeremyjaouen — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by cg-tw — DISMISSED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by cg-tw — DISMISSED** [review]

**Review by kduret — APPROVED** [review]

**Review by adr-mo — APPROVED** [review]

**Review by cg-tw — APPROVED** [review]

**Pull request merged**